### PR TITLE
Feature of having with_user_activity=True parameter when using User Activity Feed  (User Stream) url

### PR DESCRIPTION
**Pull request by ehsabd**
I liked to have a feature that I can get the user own activity by passing ``with_user_activity=True`` through query parameter like ``/activity/feed/json/with_user_activity=true``. 

So I added the ``items`` function to ``UserActivityMixin``. Take a look at the code and tell me your comments about that or this feature.

### CODE REVIEW

**Comment by ehsabd** [comment]
I guess the change I've made, while works for Json activity feed breaks Atom activity feed. So perhaps I should change another function or create a Mixin for including ``with_user_activity`` as a kwarg when calling the stream.

**Comment by ehsabd** [comment]
I have updated the code. I added ``get_stream_kwargs`` as an abstract method in ``AbstractActivityStream`` and changed the code in the ``serialize`` method of ``JSONActivityFeed`` that gets the items to make use of ``get_stream_kwargs``.

**Comment by ehsabd** [comment]
It seemed none of the solutions above fit. I finally created another mixin that I only use with ``UserJSONActivityFeed`` this prevents breaking any other classes that inherit ``UserActivityMixin`` or ``JSONActivityFeed``.

File: actstream/feeds.py [hunk 1]
```diff
@@ -242,6 +242,11 @@ def get_object(self, request, content_type_id, object_id):
     def get_stream(self):
         return any_stream
 
+class StreamKwargsMixin(object):
```
Comment (auvipy) [inline actstream/feeds.py:245]: explicit inheritance from the object is not needed.

**Review by auvipy — CHANGES_REQUESTED** [review]
also, the mixin class needed to be tested

File: actstream/feeds.py (hunk 1 above)
Comment (ehsabd) [reply actstream/feeds.py:245]: Yeah there is no need to do that. But I followed the pattern of other classes in ``feeds.py`` like ``AbstractActivityStream``, ``ModelActivityMixin``, etc.

Also I will add tests for ``StreamKwargsMixin``

**Review by auvipy — CHANGES_REQUESTED** [review]
looking for unit test

**Comment by ehsabd** [comment]
> looking for unit test

I'm working on it. I hope I can provide it soon.

**Comment by ehsabd** [comment]
I've added the related tests to ``test_feeds.py``. Indeed the json feed were not tested. So I've added a test for that and also a test with ``with_user_activity=true``.

**Comment by ehsabd** [comment]
> looking for unit test

Unit tests added. Can u take  alook?

File: docs/source/feeds.rst [hunk 2]
```diff
@@ -26,6 +26,14 @@ Shows user stream for currently logged in user.
     /activity/feed/atom/
     /activity/feed/json/
 
+If you want to include the user own activity, the optional parameter ``with_user_activity`` can be passed to the user stream as a query string parameter:
```
Comment (auvipy) [inline docs/source/feeds.rst:29]: I am willing to know if this is part of activity stream specification?

Comment (justquick) [reply docs/source/feeds.rst:29]: meh

**Pull request merged**